kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutlinePanel.js

```diff
@@ -0,0 +1,208 @@
+/* eslint-env browser */
+/**
+ * Widget for Narrative Outline Panel
+ * @author David Lyon  <dlyon@lbl.gov>
+ * @public
+ */
+define(['jquery', 'base/js/namespace', 'kbwidget', 'kbaseNarrativeControlPanel'], (
+    $,
+    Jupyter,
+    KBWidget,
+    ControlPanel
+) => {
+    'use strict';
+    return new KBWidget({
+        name: 'kbaseNarrativeOutlinePanel',
+        parent: ControlPanel,
+        version: '1.0.0',
+        options: {
+            /** If true, all markdown H1-H3 will be outline items
+             * If false, only the first header will be an outline item */
+            showAllHeaders: false,
+            /** Header height for visibility & scroll functionality */
+            headerHeight: 80,
+            /** Refresh throttle rate in ms */
+            refreshRate: 100,
+        },
+
+        isActive: false,
+        body: undefined,
+        drawnState: '', // string representing the items drawn used to eliminate excess dom manipulation
+
+        init: function (options) {
+            this._super(options);
+            this.body = $(this.body());
+
+            const refresh = this.refresh.bind(this);
+            // Trigger refresh on the following events
+            $([Jupyter.events]).on('notebook_loaded.Notebook', refresh);
+            $([Jupyter.events]).on('notebook_saved.Notebook', refresh);
+            $([Jupyter.events]).on('notebook_renamed.Notebook', refresh);
+            $([Jupyter.events]).on('command_mode.Notebook', refresh); // Clicking out of a cell
+            $([Jupyter.events]).on('create.Cell', refresh);
+            $([Jupyter.events]).on('delete.Cell', refresh);
+            $([Jupyter.events]).on('select.Cell', refresh); // Triggers on cell move
```

Comment:
Turn the events into an array and iterate through them. If it's a jQuery event handler you may be able to do the assignment in a single line with space-separated event names.